v2/kotlin-sdk/app/src/main/java/com/mongodb/app/ItemActivity.kt

```diff
@@ -0,0 +1,482 @@
+    override fun onOptionsItemSelected(menuItem: MenuItem): Boolean {
+        return when (menuItem.itemId) {
+            R.id.action_logout -> {
+                CoroutineScope(Dispatchers.IO).launch {
+                    runCatching {
+                        realmApp.currentUser?.logOut()
+                    }.onSuccess {
+                        Log.v(TAG(), "user logged out")
+                        withContext(Dispatchers.Main) {
+                            startActivity(Intent(application, LoginActivity::class.java))
+                        }
+                    }.onFailure { ex: Throwable ->
+                        Log.e(TAG(), "log out failed! Error: ${ex.message}")
+                    }
+                }
+                true
+            }
+            R.id.action_airplane_mode -> {
+                if (connectivityEnabled) {
+                    connectivityEnabled = false
+                    airplaneModeItem.icon = syncDisabledIcon
+                    realm.syncSession.pause()
+                } else {
+                    connectivityEnabled = true
+                    airplaneModeItem.icon = syncEnabledIcon
+                    realm.syncSession.resume()
+                }
+                true
+            }
+            else -> {
+                super.onOptionsItemSelected(menuItem)
+            }
+        }
+    }
+
+    /**
+     * Ensure the user realm closes when the activity ends.
+     */
+    override fun onDestroy() {
+        super.onDestroy()
+        realm.close()
+        recyclerView.adapter = null
+    }
+
+    /**
+     *  Creates a popup that allows the user to insert a item into the realm.
+     */
+    private fun onFabClicked() {
+        val dialogBuilder = AlertDialog.Builder(this)
+            .setMessage("Enter Item Name:")
+        val view: View = layoutInflater.inflate(R.layout.create_item_dialog, null)
+
+        val itemSummaryInput = view.findViewById<View>(R.id.plain_text_input) as EditText
+
+        dialogBuilder
+            .setPositiveButton("Create") { _, _ ->
+                run {
+                    val item = Item(realmApp.currentUser!!.id)
+                    item.summary = itemSummaryInput.text.toString()
+
+                    CoroutineScope(Dispatchers.IO).launch {
+                        realm.write {
+                            this.copyToRealm(item)
+                        }
+                    }
+
+                    // Display the item created using Android's Toast feedback popup
+                    Toast.makeText(
+                        this,
+                        "Item created: ${item.summary}",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            }
+            .setNegativeButton("Cancel") { dialog, _ ->
+                dialog.cancel()
+            }
+
+        dialogBuilder.setView(view)
+        val dialog = dialogBuilder.create()
+        dialog.setTitle("Add Item")
+        dialog.show()
```

Comment:
Do we intend to keep both Compose and "classic" together in the app?

Reply:
Legacy code, it has been removed.